cadc-download-manager-server/src/main/java/ca/nrc/cadc/dlm/server/PackageServlet.java

```diff
@@ -0,0 +1,122 @@
+    private static final String REQUEST_PARAM_URI_KEY = "tuple";
+    private static final String REQUEST_PARAM_METHOD_KEY = "method";
+    private static final Map<String, String> METHOD_TO_CONTENT_TYPE_MAP = new HashMap<>();
+
+    private final ApplicationConfiguration configuration = new ApplicationConfiguration(DispatcherServlet.DEFAULT_CONFIG_FILE_PATH);
+
+    static {
+        PackageServlet.METHOD_TO_CONTENT_TYPE_MAP.put(DispatcherServlet.TAR_PACKAGE, "application/x-tar");
+        PackageServlet.METHOD_TO_CONTENT_TYPE_MAP.put(DispatcherServlet.ZIP_PACKAGE, "application/zip");
+    }
+
+
+    /**
+     * Handle POSTed download request from an external page.
+     *
+     * @param request  The HTTP Request
+     * @param response The HTTP Response
+     */
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) {
+        final URI packageServiceURI = this.configuration.lookupServiceURI(PackageServlet.PACKAGE_SERVICE_RESOURCE_ID_KEY, null);
+        if (packageServiceURI == null) {
+            throw new IllegalStateException("No service configured to perform Package Downloads");
+        } else {
+            doRedirect(request, response, packageServiceURI);
+        }
+    }
+
+    void doRedirect(final HttpServletRequest request, final HttpServletResponse response, final URI packageServiceURI) {
+        final Subject currentSubject = AuthenticationUtil.getCurrentSubject();
+        final URL packageServiceURL = lookupServiceURL(packageServiceURI, currentSubject);
+
+        Subject.doAs(currentSubject, (PrivilegedAction<Void>) () -> {
+            final HttpPost httpPost = new HttpPost(packageServiceURL, getPayload(request), false);
+
+            try {
+                httpPost.prepare();
+            } catch (Exception exception) {
+                throw new RuntimeException(exception.getMessage(), exception);
+            }
+
+            final URL redirect = httpPost.getRedirectURL();
+            if (redirect == null) {
+                throw new IllegalArgumentException("Unable to complete download (no redirect).");
+            } else {
+                try {
+                    response.sendRedirect(redirect.toExternalForm());
+                } catch (IOException ioException) {
+                    throw new IllegalStateException("Redirect is invalid: " + redirect);
+                }
+            }
+
+            return null;
+        });
+    }
+
+    private Map<String, Object> getPayload(final HttpServletRequest request) {
+        final Map<String, Object> payload = new HashMap<>();
+        final Map<String, String[]> requestParameters = request.getParameterMap();
+
+        final String[] publisherIDs = requestParameters.get(PackageServlet.REQUEST_PARAM_URI_KEY);
+        if (publisherIDs == null || publisherIDs.length == 0) {
+            throw new IllegalArgumentException("Nothing specified to download.  Use tuple=<URI>.");
```

Comment:
Minor comment: could replace `tuple` with `PackageServlet.REQUEST_PARAM_URI_KEY`. It applies to the other ones too.